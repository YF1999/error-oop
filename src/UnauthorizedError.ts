import { IErrorOptions, IErrorArguments, NativeError, ErrorTool } from './Native';

export interface IUnauthorizedErrorOptions extends IErrorOptions {}

/**
 * Applicable when the operating system denies access because of an I/O error or a specific type of security error.
 */
export class UnauthorizedError extends NativeError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be appended.
     */
    public constructor(message: string, innerError: Error);
    /**
     * @param options The constructor options.
     */
    public constructor(options: IUnauthorizedErrorOptions);

    public constructor(...args: IErrorArguments<IUnauthorizedErrorOptions>) {
        super(ErrorTool.parseErrorArguments([], ...args));
    }
}
