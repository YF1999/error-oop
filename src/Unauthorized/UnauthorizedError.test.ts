import { UnauthorizedError } from './UnauthorizedError';

standardTestError(UnauthorizedError, 'UnauthorizedError', 'error');
