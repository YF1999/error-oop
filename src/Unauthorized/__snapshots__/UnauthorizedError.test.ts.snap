// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`UnauthorizedError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`UnauthorizedError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`UnauthorizedError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`UnauthorizedError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
